integration-tests/appsec/endpoints-collection.spec.js

```diff
@@ -0,0 +1,151 @@
+  let sandbox, cwd
+
+  before(async function () {
+    this.timeout(process.platform === 'win32' ? 90000 : 30000)
+
+    sandbox = await createSandbox(
+      ['fastify'],
+      false
+    )
+
+    cwd = sandbox.folder
+  })
+
+  after(async function () {
+    this.timeout(60000)
+    await sandbox.remove()
+  })
+
+  function getExpectedEndpoints (framework) {
+    const expectedEndpoints = [
+      // Basic routes
+      { method: 'GET', path: '/users' },
+      { method: 'HEAD', path: '/users' },
+      { method: 'POST', path: '/users/' },
+      { method: 'PUT', path: '/users/:id' },
+      { method: 'DELETE', path: '/users/:id' },
+      { method: 'PATCH', path: '/users/:id/:name' },
+      { method: 'OPTIONS', path: '/users/:id?' },
+
+      // Route with regex
+      { method: 'DELETE', path: '/regex/:hour(^\\d{2})h:minute(^\\d{2})m' },
+
+      // Additional methods
+      { method: 'TRACE', path: '/trace-test' },
+      { method: 'HEAD', path: '/head-test' },
+
+      // Custom method
+      { method: 'MKCOL', path: '/example/near/:lat-:lng/radius/:r' },
+
+      // Using app.route()
+      { method: 'POST', path: '/multi-method' },
+      { method: 'PUT', path: '/multi-method' },
+      { method: 'PATCH', path: '/multi-method' },
+
+      // All supported methods route
+      { method: 'GET', path: '/all-methods' },
+      { method: 'HEAD', path: '/all-methods' },
+      { method: 'TRACE', path: '/all-methods' },
+      { method: 'DELETE', path: '/all-methods' },
+      { method: 'OPTIONS', path: '/all-methods' },
+      { method: 'PATCH', path: '/all-methods' },
+      { method: 'PUT', path: '/all-methods' },
+      { method: 'POST', path: '/all-methods' },
+      { method: 'MKCOL', path: '/all-methods' }, // Added with addHttpMethod
+
+      // Nested routes with Router
+      { method: 'PUT', path: '/v1/nested/:id' },
+
+      // Deeply nested routes
+      { method: 'GET', path: '/api/nested' },
+      { method: 'HEAD', path: '/api/nested' },
+      { method: 'GET', path: '/api/sub/deep' },
+      { method: 'HEAD', path: '/api/sub/deep' },
+      { method: 'POST', path: '/api/sub/deep/:id' },
+
+      // Wildcard routes
+      { method: 'GET', path: '/wildcard/*' },
+      { method: 'HEAD', path: '/wildcard/*' },
+      { method: 'GET', path: '*' },
+      { method: 'HEAD', path: '*' },
+
+      { method: 'GET', path: '/later' },
+      { method: 'HEAD', path: '/later' },
+    ]
+
+    return expectedEndpoints
+  }
+
+  async function runEndpointTest (framework) {
+    let agent, proc
+    const appFile = path.join(cwd, 'appsec', 'endpoints-collection', `${framework}.js`)
+
+    try {
+      agent = await new FakeAgent().start()
+
+      const expectedEndpoints = getExpectedEndpoints(framework)
+      const endpointsFound = []
+      const isFirstFlags = []
+
+      const telemetryPromise = agent.assertTelemetryReceived(msg => {
+        const { payload } = msg
+        if (payload.request_type === 'app-endpoints') {
```

Comment:
I think this conditional is useless, since the check is already done in fake agent:
https://github.com/DataDog/dd-trace-js/blob/master/integration-tests/helpers/fake-agent.js#L214

Reply:
BAM [`29ef6ef` (#6258)](https://github.com/DataDog/dd-trace-js/pull/6258/commits/29ef6ef1feea7ece950919f04272f018a120f7bf)